paymentsheet/src/test/java/com/stripe/android/paymentsheet/ui/PaymentSheetTopBarTest.kt

```diff
@@ -0,0 +1,112 @@
+package com.stripe.android.paymentsheet.ui
+
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.platform.LocalSoftwareKeyboardController
+import androidx.compose.ui.platform.SoftwareKeyboardController
+import androidx.compose.ui.test.performClick
+import androidx.compose.ui.unit.dp
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import com.stripe.android.paymentsheet.R
+import com.stripe.android.utils.HackyComposeTestRule
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.never
+import org.mockito.kotlin.verify
+
+@OptIn(ExperimentalComposeUiApi::class)
+@RunWith(AndroidJUnit4::class)
+class PaymentSheetTopBarTest {
+
+    @get:Rule
+    val composeTestRule = HackyComposeTestRule()
+
+    @Test
+    fun `Handles navigation icon press correctly`() {
+        val mockKeyboardController = mock<SoftwareKeyboardController>()
+        var didCallOnNavigationIconPressed = false
+
+        composeTestRule.setContent {
+            CompositionLocalProvider(
+                LocalSoftwareKeyboardController provides mockKeyboardController,
+            ) {
+                PaymentSheetTopBar(
+                    state = mockState(isEnabled = true),
+                    elevation = 0.dp,
+                    onNavigationIconPressed = { didCallOnNavigationIconPressed = true },
+                    onEditIconPressed = {},
```

Comment:
Can we add a throw here for the actions we don't expect?

```suggestion
                    onEditIconPressed = { throw AssertionError("Not Expected") },
```